src/js/utils/RouterUtil.js

```diff
@@ -17,6 +17,14 @@ function findRedirect(queryString) {
 }
 
 const RouterUtil = {
+  isValidRedirect(domain) {
+    if (!Util.isString(domain)) {
+      return false;
+    }
+
+    return domain.includes(global.location.hostname);
```

Comment:
Then here we check if the whole thing we've provided _includes_ `location.hostname`

Imagine this:
`global.location.hostname === "my-cloud.acme.com"`
`domain === "http://malicious.domain.com/pwned?my-cloud.acme.com"`